app/js/ProjectController.js

```diff
@@ -0,0 +1,53 @@
+const logger = require('logger-sharelatex')
+const ProjectHandler = require('./ProjectHandler')
+const { expressify } = require('./Utils')
+
+module.exports = {
+  home,
+  projects: expressify(projects),
+  getProject: expressify(getProject)
+}
+
+function home(req, res, next) {
+  res.redirect('/project')
+}
+
+async function projects(req, res, next) {
+  const { user } = res.locals
+  logger.log({ userId: user._id.toString() }, 'showing project page')
+  const projects = await ProjectHandler.getUserProjects(user)
+  res.render('projects', { projects })
+}
+
+async function getProject(req, res, next) {
+  const { user } = res.locals
+  const { projectId } = req.params
+  logger.log({ projectId }, 'downloading project')
+  const project = await ProjectHandler.getProject(projectId)
+  if (project == null) {
```

Comment:
I think these checks for the user owning the project ought to be in their own authorization middleware `userOwnsProject`.  You can store the project in `req.projects` so that it's available in `ProjectController.getProject`.

Reply:
I started in that direction, then I thought that it would be simpler to just not make a difference between existing and non-existent projects. Once you're logged in, a request for any project except your own returns a 404. See PR review commit below. What do you think?

Reply:
Yes your new way is better. Good.